frontend/javascripts/admin/tasktype/task_type_create_view.tsx

```diff
@@ -43,464 +62,448 @@ function isValidMagnification(_rule, value) {
+  if (value && value > minMag) {
+    return Promise.resolve();
+  }
+  return Promise.reject(
+    new Error("The maxiumum resolution needs to be larger then the minimum mag."),
+  );
 }
 
-class TaskTypeCreateView extends React.PureComponent<Props, State> {
-  formRef = React.createRef<FormInstance>();
-  state: State = {
-    teams: [],
-    useRecommendedConfiguration: false,
-    isFetchingData: false,
-  };
+function TaskTypeCreateView({ taskTypeId, history }: Props) {
+  const [useRecommendedConfiguration, setUseRecommendedConfiguration] = useState(false);
+  const [isFetchingData, setIsFetchingData] = useState(false);
+  const [form] = Form.useForm<FormValues>();
 
-  componentDidMount() {
-    this.fetchData();
-    this.applyDefaults();
-  }
+  const teams = useFetch(
+    async () => {
+      const teams = await getEditableTeams();
+      setIsFetchingData(false);
+      return teams;
+    },
+    [],
+    [],
+  );
+
+  useEffect(() => {
+    applyDefaults();
+  }, []);
+
+  async function applyDefaults() {
+    const taskType = taskTypeId ? await getTaskType(taskTypeId) : null;
 
-  async applyDefaults() {
-    const taskType = this.props.taskTypeId ? await getTaskType(this.props.taskTypeId) : null;
-    const hasRecommendedConfiguration = taskType?.recommendedConfiguration != null;
     const defaultValues = {
+      isResolutionRestricted: false,
       settings: {
         somaClickingAllowed: true,
         branchPointsAllowed: true,
         volumeInterpolationAllowed: false,
         mergerMode: false,
-        preferredMode: null,
-        resolutionRestrictions: {},
+        resolutionRestrictions: {
+          min: 1,
+          max: 512,
+        },
       },
-      recommendedConfiguration: hasRecommendedConfiguration
-        ? {}
-        : getDefaultRecommendedConfiguration(),
+      recommendedConfiguration: jsonStringify(getDefaultRecommendedConfiguration()),
     };
 
     // Use merge which is deep _.extend
-    const formValues = _.merge({}, defaultValues, taskType);
-
-    if (!hasRecommendedConfiguration) {
-      // A recommended configuration of null overrides the default configuration when using _.merge
-      // If the task type has no recommended configuration, suggest the default one
-      formValues.recommendedConfiguration = defaultValues.recommendedConfiguration;
-    }
-
-    // @ts-expect-error ts-migrate(2540) FIXME: Cannot assign to 'recommendedConfiguration' becaus... Remove this comment to see the full error message
-    formValues.recommendedConfiguration = jsonStringify(formValues.recommendedConfiguration);
-    // The format of settings.resolutionRestrictions does not match the form precisely.
-    // It is replaced here by resolutionRestrictionsForm, wich has a shouldRestrict boolean
-    // @ts-expect-error ts-migrate(2339) FIXME: Property 'resolutionRestrictionsForm' does not exi... Remove this comment to see the full error message
-    formValues.settings.resolutionRestrictionsForm = {
```

Comment:
I didn't like that the interface for `settings.resolutionRestrictions` is slightly different between this form and a task type instances. (i.e. one has a `shouldRestrict` property). 

I moved the `shouldRestrict` value out of the `settings` into it's own form value `isResolutionRestricted`(see line 105ff), so the interfaces match once again. This safes a lot of converting back and forth.